src/llmq/signing_shares.cpp

```diff
@@ -196,25 +196,45 @@ CSigSharesManager::CSigSharesManager(CConnman& connman, CChainState& chainstate,
 
 CSigSharesManager::~CSigSharesManager() = default;
 
-void CSigSharesManager::StartWorkerThread()
+void CSigSharesManager::Start()
 {
-    // can't start new thread if we have one running already
-    if (workThread.joinable()) {
+    // can't start if threads are already running
+    if (housekeepingThread.joinable() || dispatcherThread.joinable()) {
         assert(false);
     }
 
-    workThread = std::thread(&util::TraceThread, "sigshares", [this] { WorkThreadMain(); });
+    // Initialize worker pool
+    int workerCount = std::clamp(static_cast<int>(std::thread::hardware_concurrency() / 2), 1, 4);
+    workerPool.resize(workerCount);
+    RenameThreadPool(workerPool, "sigsh-work");
+
+    // Start housekeeping thread
+    housekeepingThread = std::thread(&util::TraceThread, "sigsh-maint",
+        [this] { HousekeepingThreadMain(); });
+
+    // Start dispatcher thread
+    dispatcherThread = std::thread(&util::TraceThread, "sigsh-dispatch",
```

Comment:
nit: I am not sure, but thread names likely is limited to 12 bytes, aren't they?